### ISSUE TITLE
Shopify App Review reports of a 3rd-party cookie. I traced it and it came from koa-shopify-auth

### ISSUE THREAD

**Issue by ardeearam** [Shopify/koa-shopify-auth#81]
# Issue summary

Shopify App Review reports of a 3rd-party cookie. I traced it and it came from koa-shopify-auth

![image](https://user-images.githubusercontent.com/1546228/112759906-b1708e00-9027-11eb-9f1d-0c9877b1a979.png)

![image](https://user-images.githubusercontent.com/1546228/112759930-c6e5b800-9027-11eb-9580-7ca5b200f7b6.png)

## Expected behavior

koa-shopify-auth should NOT use 3rd-party cookies, since the Shopify QA Team forbids it on your app.

## Actual behavior

koa-shopify-auth uses 3rd-party cookies AND indexed database, and thus resulting in your app getting returned for update.

Tip: include an error message (in a `<details></details>` tag) if your issue is related to an error

## Steps to reproduce the problem

1. Open an app in Incognito Google Chrome
1. Use the "eye" tool of Google Chrome to check for 3rd-party cookies coming from `koa-shopify-auth`
![image](https://user-images.githubusercontent.com/1546228/112759973-ed0b5800-9027-11eb-8576-c9490d12f072.png)

## Reduced test case

The best way to get your bug fixed is to provide a [reduced test case](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Reducing_testcases).

---

## Checklist

- [X] Please delete the labels section before submitting your issue
- [X] I have described this issue in a way that is actionable (if possible)

**Comment by paulomarg** [comment]
Thanks for reporting this @ardeearam, we'll look into it!

**Comment by Anfevazu** [comment]
@paulomarg Hello Paul, We are having the same issue and we are using the official Shopify library. It have been 2 rejections for this same issue, please your help.

**Comment by pratiknikam** [comment]
Make sure to clear your iFrame cookies after making this change otherwise you would get:

`Cannot complete OAuth process. Could not find an OAuth cookie for shop url: myshop.myshopify.com`

**Comment by pratiknikam** [comment]
@paulomarg not sure if this is mentioned on any other issues but after this upgrade, there is intermittent enable cookies screen that pop-up whenever app is selected from apps section. This shows only for fraction of a second. Is that normal?

![image](https://user-images.githubusercontent.com/2853953/113645706-653fd200-9655-11eb-9218-6d082a1ed155.png)

**Comment by paulomarg** [comment]
Hey @pratiknikam, that behaviour is expected - that page is used to call `requestStorageAccess` when the `Enable cookies` button is clicked, which is needed (especially on Safari) to be able to use cookies. Once that permission is given, the page will detect that it already has storage access and move on, so it doesn't have to wait for the user to click it.

**Comment by mcunningham** [comment]
Hey @paulomarg, I am also getting an intermittent enable cookies screen similar to @pratiknikam when using session tokens in my embedded app. It flashes for a second and then auth process is re-triggered in the background. Just to clarify - while you mentioned this is expected behavior this seems to happen at seemingly random times for me. Is there a more specific  and reproducible reason that this screen gets triggered? For me it has happened after I select my app in the app section but also had it happen when a html/react form was submitted in my app. Both of these times were after initial auth. 

Could this be due to the session not loading properly in a development environment? Would a production app with proper [session storage](https://github.com/Shopify/shopify-node-api/blob/main/docs/issues.md#notes-on-session-handling) fix the occurrences of this? 

**TL;DR** I am having intermittent occurrences of the enable cookies screen while using session tokens and I am unable to track down what is causing this screen to trigger so frequently

**Comment by redareda9** [comment]
My app is getting declined because of this exact problem of the intermittent cookie screen. 
@paulomarg Is the project abandoned as people are not getting any answers in this repo for few months? It would be good to give a disclaimer before we all start to use koa-shopify-auth thinking it respects Shopify standards of quality and then realizing after days/weeks/months of coding that we need to use something else so that the application can get approved.

**Comment by rbmdotdev** [comment]
This ticket should be reopened imo, this is a recurring problem that impacts both the review process and the usability of the app.  Our users are constantly having to click enable cookies particularly on ipad within POS

**Comment by TheSecurityDev** [comment]
Guys please check out my version of this library that doesn't use cookies and fixes this popup.

https://www.npmjs.com/package/simple-koa-shopify-auth

Note: I think it works in Safari, but I'm not 100% sure.
